src/containers/builder/builder.container.tsx

```diff
@@ -0,0 +1,43 @@
+import { Button, Icon } from 'antd';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { StatusTag } from '../../common/components/status-tag/status-tag.component';
+import { Survey } from '../../common/types/survey.type';
+import { AppState } from '../../state';
+import './builder.container.less';
+
+export interface BuilderRouteInfo {
+    surveyid: string;
+}
+
+export interface BuilderContainerProps {
+    survey: Survey;
+}
+
+const mapStateToProps = (state: AppState): Partial<BuilderContainerProps> => ({
+    survey: state.surveysState.surveys.find(s => s.id === state.router.location.pathname.split('/').reverse()[0]),
```

Comment:
this needs to become a helper function...

Reply:
I think this can be done through `state.match.params.surveyId` or something like it